### PR TITLE
Storybook Refactor

### PR DESCRIPTION
**Pull request by nicklathe**
Part of the work upgrade Storybook. This PR addresses the following:

* DeleteAccountDialog.story.jsx
* AddParentEmailModal.story.jsx
* ChangeEmailModal.story.jsx
* ChangeUserTypeForm.story.jsx
* ChangeUserTypeModal.story.jsx

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/lib/ui/accounts/ChangeUserTypeModal.story.jsx [hunk 1]
```diff
@@ -2,12 +2,26 @@ import React from 'react';
...
-  storybook
-    .storiesOf('ChangeUserTypeModal', module)
-    .add('student becoming teacher', () => (
-      <ChangeUserTypeModal
-        handleSubmit={action('handleSubmit callback')}
-        handleCancel={action('handleCancel callback')}
-      />
-    ));
+const DEFAULT_PROPS = {
+  handlSubmit: action('handleSubmit callback'),
```
Comment (mgc1194) [inline apps/src/lib/ui/accounts/ChangeUserTypeModal.story.jsx]: typo in `handleSubmit`

File: apps/src/lib/ui/accounts/AddParentEmailModal.story.jsx [hunk 2]
```diff
@@ -2,19 +2,32 @@ import React from 'react';
...
-    .add('view with current parent email', () => (
-      <AddParentEmailModal
-        handleSubmit={action('handleSubmit callback')}
-        handleCancel={action('handleCancel callback')}
-        currentParentEmail={'minerva@hogwarts.edu'}
-      />
-    ));
+const DEFAULT_PROPS = {
+  handleSubmit: action('handleSubmit callback'),
+  handleCancel: action('handlCancel callback')
```
Comment (wilkie) [inline apps/src/lib/ui/accounts/AddParentEmailModal.story.jsx]: Typo here, as well.

**Comment by wilkie** [comment]
Beyond some typos, looks good overall. Thank you for taking the time to do a few of these!

File: apps/src/lib/ui/accounts/ChangeUserTypeModal.story.jsx (hunk 1 above)
Comment (nicklathe) [reply apps/src/lib/ui/accounts/ChangeUserTypeModal.story.jsx]: ✅

File: apps/src/lib/ui/accounts/AddParentEmailModal.story.jsx (hunk 2 above)
Comment (nicklathe) [reply apps/src/lib/ui/accounts/AddParentEmailModal.story.jsx]: ✅

File: apps/src/lib/ui/accounts/ChangeUserTypeForm.story.jsx [hunk 3]
```diff
@@ -4,59 +4,50 @@ import {action} from '@storybook/addon-actions';
 
 const DEFAULT_PROPS = {
   values: {
-    currentEmail: '',
+    email: '',
```
Comment (nicklathe) [inline apps/src/lib/ui/accounts/ChangeUserTypeForm.story.jsx:7]: The way this existed before was broken, this key should be `email` as you can [see here](https://github.com/code-dot-org/code-dot-org/blob/685ba29daaba2cd51c872267e2ca02d1d1b920a7/apps/src/lib/ui/accounts/ChangeUserTypeForm.jsx#L10), not `currentEmail`

**Comment by nicklathe** [comment]
I've addressed the two typos as well as a couple other minor cleanup items, and fixed an already existing bug. Ready for a final pass.

File: apps/src/lib/ui/accounts/ChangeUserTypeForm.story.jsx (hunk 3 above)
Comment (wilkie) [reply apps/src/lib/ui/accounts/ChangeUserTypeForm.story.jsx:7]: Wow. I wonder why it worked before. Perhaps this API change suddenly paid for itself.

**Review by wilkie — APPROVED** [review]
LGTM 👍

**Pull request merged**